### PR TITLE
[DRAFT] Add support for start -i (help required)

### LINKED ISSUE #3103: `nerdctl start` does not support interactive
### Description

Docker does.

### Steps to reproduce the issue

na

### Describe the results you received and expected

na

### What version of nerdctl are you using?

1.7.6

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

_No response_

### PR DESCRIPTION
**Pull request by apostasie**
Hey folks.

This is to tentatively fix #3103

I am slightly out of my comfort zone here and can use some advice and thorough review here, especially on the combination behavior of --attach, --stdin and what that means for detaching or not.

Sending as draft for now for that reason.

### CODE REVIEW

**Comment by apostasie** [comment]
CI is green.

@AkihiroSuda and other good folks here, PTAL at your convenience and lmk if I got this one right.

**Comment by AkihiroSuda** [comment]
Can we have an integration test to verify this?

**Comment by apostasie** [comment]
> Can we have an integration test to verify this?

Of course. Coming later today or tomorrow.

File: cmd/nerdctl/container_start.go [hunk 1]
```diff
@@ -39,6 +39,7 @@ func newStartCommand() *cobra.Command {
 	startCommand.Flags().SetInterspersed(false)
 	startCommand.Flags().BoolP("attach", "a", false, "Attach STDOUT/STDERR and forward signals")
 	startCommand.Flags().String("detach-keys", consoleutil.DefaultDetachKeys, "Override the default detach keys")
+	startCommand.Flags().BoolP("interactive", "i", false, "Keep STDIN open even if not attached")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_start.go:42]: Please update docs/command-reference.md too

**Comment by apostasie** [comment]
@AkihiroSuda test added.

New test is green locally, but let's wait for the CI.

PTAL at your convenience.

File: cmd/nerdctl/container_start_linux_test.go [hunk 2]
```diff
@@ -61,3 +61,45 @@ func TestStartDetachKeys(t *testing.T) {
...
+
+	cmd := base.Cmd("container", "start", "-i", containerName).
+		CmdOption(testutil.WithStdin(strings.NewReader("echo " + needle)))
+
+	// XXX docker behavior does not make much sense here
+	if base.Target == testutil.Docker {
+		cmd.AssertOutExactly("")
+	} else {
+		cmd.AssertOutExactly(needle + "\n")
+	}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container_start_linux_test.go:94]: Can we just follow Docker's behavior, even if it does not make much sense?

Comment (apostasie) [reply cmd/nerdctl/container_start_linux_test.go:94]: Of course. Let me look how to do that.

File: cmd/nerdctl/container_start.go (hunk 1 above)
Comment (apostasie) [reply cmd/nerdctl/container_start.go:42]: ack.
always forget these... o`'o

**Comment by apostasie** [comment]
@AkihiroSuda this is a lot more complex than I thought.

The combination matrix of `create [--tty] [--interactive]` and `start [--interactive] [--attach]` has significant subtleties and console juggling that requires more research on my side.

I am converting this back to draft for now.

**Comment by apostasie** [comment]
@AkihiroSuda I don't think I can put more work in this in time for v2.

Feel free to remove it from the milestone if you prefer.

**Comment by apostasie** [comment]
I am going to close this for the time being to keep the tracker cleaner - will reopen when I have the ability to focus on it.

**Pull request closed without merging**